src/corehost/cli/bdl_processor.cpp

```diff
@@ -0,0 +1,316 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifdef FEATURE_APPHOST
+
+#include "bdl_processor.h"
+#include "pal.h"
+#include "trace.h"
+#include "utils.h"
+
+void bdl_processor_t::seek(long offset, int origin)
+{
+	if (fseek(m_bundle, offset, origin) != 0)
+	{
+		trace::error(_X("Bundle extraction: Seek failure."));
+		throw StatusCode::BundleExtractionIOError;
```

Comment:
Please do not use exceptions.

Reply:
@vitek-karas  Can you please expand on why you're suggesting not to use exceptions? 
There are a couple of exception-throws in [existing code](https://github.com/dotnet/core-setup/blob/18780678da576d8c629066f50af30159a8859b2f/src/corehost/cli/ijwhost/ijwthunk.cpp#L136-L145). And exceptions are used by json-parser as well.

Reply:
hmm - I missed the exceptions in the IJW... 😞 
The json parser is basically a necessary evil - we took that code from elsewhere.

The reason I don't like exceptions in this kind of code is that it's very easy to forget and let the exception escape. Which will that wreak havoc on the caller - specifically since it's a C++ exception, so a very weird thing for any language but C++.

The return codes are a bit more messy to work with, but they make error paths very explicit.

It was also my impression that the code base doesn't use them - so for consistency.

I'm open to discussion, but I think we should have a good reason to start using them in general.

Reply:
>> The return codes are a bit more messy to work with, but they make error paths very explicit.

Definitely. I started out with propagating return codes, but the source code was filled with error handling `if-else` checks -- which (subjectively) seemed to obscure the overall code flow. 
The bundle-extraction logic is IO-intensive, and has a higher density of error checks.

>> The reason I don't like exceptions in this kind of code is that it's very easy to forget and let the 
exception escape. 

Conversely, I think it is even more easy to forget checking or propagating an error code leading to failure/unexpected behavior in an unrelated part of the code. 

>> I'm open to discussion, ... 

How about we take a middle ground, where we say that:
* We only throw a Status code exceptions 
* We only throw to indicate critical failure.
* Each subsystem (ex: bundle-handler, deps-resolver) catches its own exceptions. 

I can certainly change the code back to using if-else checks, if we're still not convinced.

Reply:
OK - I guess I'll warm up to it eventually. I'm fine if there's a clear entry point to the subsystem where we can easily try/catch and isolate the rest of the code from the exceptions that way.

Reply:
The IJW host exceptions are also a necessary evil because the function that can throw executes in an arbitrary call stack by code that may or may not know that it is starting up the runtime and as a result has no other way to propagate an error code.